### ISSUE TITLE
[Bug]: Hexbear instance issue with “www” part of url

### ISSUE THREAD

**Issue by soren446** [Memmy-App/memmy#941]
### Check for open issues

- [X] I have verified that another issue for this is not open, or it has been closed and has not been fixed.

### Minimal reproducible example

Logging into the hexbear.net requires you to use the www.hexbear.net url to sign into your account. This causes errors with loading communities and usernames by clicking on a user/community, as they attempt to load as “[community]@www.hexbear.net” instead of “[community]@hexbear.net. 

### Expected Behavior

Expected behavior should ignore “www.” from the account log in url when generating a page from clicking on communities or users, since some lemmy instances (like hexbear) require the “www”. Instead, when clicking on a community or username, Memmy generates a url with @“www.hexbear.net” which is not a valid url. 

### Version

0.5.1

### App Store or TestFlight?

App Store

### iOS Version

16.x

### Device

iPhone 14 Pro Max

**Comment by soren446** [comment]
![IMG_3646](https://github.com/Memmy-App/memmy/assets/141293054/b99c9402-95e3-438b-8ab3-cd16d02d407f)
![IMG_3645](https://github.com/Memmy-App/memmy/assets/141293054/5480c516-2074-4090-86da-5b07a50e6775)

Attaching here screenshots just in case.

**Comment by gkasdorf** [comment]
Took me a minute to understand what you were saying because, well, this is super weird.

I don't really agree that we should be the ones to remove www from the domains, but I do see the issue you have here. There's two issues it seems here that are not really our fault. (Note that this explanation is based on my admittedly lacking knowledge on how the configuration for activitypub goes)

1. The server is using the www subdomain for accessing their site. This seems like a pretty poor choice for a system like this. Imagine using, for example, XMPP and having to sign in to your account with `username@www.xmppchat.com` but then communicating with other users by messaging `username@xmppchat.com`.
2. The server then must be configuring activity pub to federate with `hexbear.net` which is missing the subdomain. This is a very poor choice because of, well, issues like this. Either they should federate with `www.hexbear.net` (poor choice in my opinion) or change their web server to not redirect to the `www` subdomain (much easier to do anyway, since it wouldn't affect what federation has already done).

I'd strongly advise reaching out to hexbear admins and request that change. I'm afraid that if we decide to remove the www from the URLs, we will end up with cases where some instances might actually be configured to use the www for federation and we would have issues.

Bottom line is that the instance's `Lemmy.hjson` configuration should match the web server.

**Comment by soren446** [comment]
Still learning this whole activitypub thing, but that makes sense. Thanks for clarifying! Will definitely reach out to them.

**Comment by gkasdorf** [comment]
@soren446 No problem. Let me know their response if you get one. Like I said, I believe this is a pretty significant configuration issue on their part but am curious to hear the other side.
